modules/Auth/lib/AuthHandler.cpp

```diff
@@ -124,6 +124,20 @@ TokenHandlingResult AuthHandler::on_token(const ProvidedIdToken& provided_token)
     return result;
 }
 
+void AuthHandler::handle_token_validation_result_update(const ValidationResultUpdate& validation_result_update) {
+    auto connector_id = validation_result_update.connector_id;
```

Comment:
This should be protected using the `event_mutex` since `this->evses` is accessed and updated.  
```suggestion
    std::unique_lock<std::mutex> lk(this->event_mutex);
    auto connector_id = validation_result_update.connector_id;
```